src/js/utils/captions.js

```diff
@@ -89,13 +95,14 @@ export const parseTracks = (trackOrTracks) => {
 /**
  * Translates a TextTrack-like object into a well-defined string representation for the TextTrack
  * @param {Object} obj - A TextTrack or TextTrack-like object
+ * @param {string} [obj.kind] - An optional kind for the track.
```

Comment:
question(blocker): Consider making this a required field? Looks like it will simplify a lot.